spec/support/initializers/rspec_csv_formatter.rb

```diff
@@ -0,0 +1,35 @@
+require "csv"
+
+class CSVFormatter < RSpec::Core::Formatters::JsonFormatter
+  RSpec::Core::Formatters.register self
+
+  def close(_notification)
+    headers = ["Spec", "File", "Status", "Run Time", "Exception"]
+    headers += ["Travis Branch", "Travis URL"]
+    with_headers = { write_headers: true, headers: headers }
+
+    CSV.open(output.path, "w", with_headers) do |csv|
+      @output_hash[:examples].map do |ex|
+        file_path = "#{ex[:file_path]}:#{ex[:line_number]}"
+        row = [ex[:full_description], file_path, ex[:status], ex[:run_time], ex[:exception]]
+        row += [ENV["TRAVIS_BRANCH"], ENV["TRAVIS_BUILD_WEB_URL"]]
+        csv << row
+      end
+    end
+  end
+end
+
+if ENV["TRAVIS"]
+  RSpec.configure do |config|
+    tmp_dir = File.join(Dir.pwd, "tmp")
+    FileUtils.mkdir(tmp_dir) unless File.directory?(tmp_dir)
```

Comment:
Ruby also has [`FileUtils.mkdir_p`](https://ruby-doc.org/stdlib-2.7.1/libdoc/fileutils/rdoc/FileUtils.html#method-c-mkdir_p) which behaves like `mkdir -p` in the shell, so you can simplify this a bit to

```ruby
csvs_dir = File.join("tmp", "csvs")
FileUtils.mkdir_p(csvs_dir)
```